compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -951,27 +947,48 @@ static void build_enum_cast_function(EnumType* et) {
 }
 
 
-static void build_enum_assignment_function(EnumType* et) {
-  if (function_exists("=", et, et))
+// Helper function used to create "=", "==", and "!=" for enums
+static void build_enum_binary_op(EnumType* et, const char* op,
+                                 PrimitiveTag prim) {
+  if (FnSymbol* fn = function_exists(op, et, et)) {
+    USR_FATAL_CONT(fn, "Can't override '%s' for enums", op);
     return;
+  }
+  bool isAssign = (prim == PRIM_ASSIGN);
 
-  FnSymbol* fn = new FnSymbol("=");
-  fn->addFlag(FLAG_ASSIGNOP);
+  FnSymbol* fn = new FnSymbol(op);
+  if (prim == PRIM_ASSIGN) {
+    fn->addFlag(FLAG_ASSIGNOP);
+  }
   fn->addFlag(FLAG_COMPILER_GENERATED);
-  fn->addFlag(FLAG_LAST_RESORT);
```

Comment:
I went back and forth on this question while writing the PR.  In writing it, I worried that it's more likely to result in confusion than benefit which is why I disabled it here (I also figured this would be the sort of thing that would be easier to relax later than it would be to tighten up on).  But I think you're right that it'd be consistent with what we do for other types to let users write their own comparison overloads and potentially shoot themselves in the foot.  Would you like me to put this back in?